app/services/ping.rb

```diff
@@ -63,7 +63,7 @@ def statuses_smart_proxies
         begin
           version = proxy.statuses[:version].version['version']
           features = proxy.statuses[:version].version['modules']
-          failed_features = proxy.statuses[:logs].logs.failed_modules
+          failed_features = ProxyAPI::V2::Features.new(:url => proxy.url).features.select { |f, p| p['state'] == 'failed' }
```

Comment:
I think the only downside of this is that it now does 2 HTTP calls to the features API. This can make it less reliable and certainly slower. It does show the duration in the result so that will show up to end users as well.